tests/cypress/tests/unit/edges_test.ts

```diff
@@ -0,0 +1,319 @@
+
+            expect(vertical.length).to.equal(4);
+            vertical.forEach(edges => {
+                expect(edges.length).to.equal(3);
+
+                edges.forEach((edge, index) => {
+                    expect(edge).to.equal(index === edges.length - 1 ?
+                        Environment.defaultEdge :
+                        '1px solid green'
+                    );
+                });
+            });
+        }
+    });
+
+    it('applies `borderLeft` overridden by higher precedence `borderRight`', () => {
+        const res = edgesFn(
+            [
+                { id: 'id', name: 'id' },
+                { id: 'name', name: 'name' }
+            ],
+            [{
+                style: { borderLeft: '1px solid green' },
+                matchesColumn: () => true,
+                matchesFilter: () => true,
+                matchesRow: () => true
+            }, {
+                style: { borderRight: '1px solid darkgreen' },
+                matchesColumn: () => true,
+                matchesFilter: () => true,
+                matchesRow: () => true
+            }],
+            [
+                { id: 1, name: 'a' }
+            ],
+            { columns: 0, rows: 0 }
+        );
+
+        expect(res !== undefined).to.equal(true);
+        if (res) {
+            const { vertical } = res.getEdges();
+
+            expect(vertical.length).to.equal(1);
+            expect(vertical[0].length).to.equal(3);
+            expect(vertical[0][0]).to.equal('1px solid green');
+            expect(vertical[0][1]).to.equal('1px solid darkgreen');
+            expect(vertical[0][2]).to.equal('1px solid darkgreen');
+        }
+    });
+
+    it('applies `borderLeft` not overridden by lower precedence `borderRight`', () => {
+        const res = edgesFn(
+            [
+                { id: 'id', name: 'id' },
+                { id: 'name', name: 'name' }
+            ],
+            [{
+                style: { borderRight: '1px solid darkgreen' },
+                matchesColumn: () => true,
+                matchesFilter: () => true,
+                matchesRow: () => true
+            }, {
+                style: { borderLeft: '1px solid green' },
+                matchesColumn: () => true,
+                matchesFilter: () => true,
+                matchesRow: () => true
+            }],
+            [
+                { id: 1, name: 'a' }
+            ],
+            { columns: 0, rows: 0 }
+        );
+
+        expect(res === undefined).to.equal(false);
+        if (res) {
+            const { vertical } = res.getEdges();
+
+            expect(vertical.length).to.equal(1);
+            expect(vertical[0].length).to.equal(3);
+            expect(vertical[0][0]).to.equal('1px solid green');
+            expect(vertical[0][1]).to.equal('1px solid green');
+            expect(vertical[0][2]).to.equal('1px solid darkgreen');
+        }
+    });
+
+    it('applies `borderLeft` overridden by higher precedence `border`', () => {
```

Comment:
can we test the opposite case too, `borderLeft` overriding only the left with `border` still setting the other three sides

Reply:
adding